packages/client/hmi-client/src/components/workflow/tera-workflow.vue

```diff
@@ -718,12 +718,37 @@ function removeEdges(portId: string) {
 	const edges = wf.value.edges.filter(
 		({ targetPortId, sourcePortId }) => targetPortId === portId || sourcePortId === portId
 	);
+
+	// Build a traversal map before we do actual removal
+	const nodeMap = new Map<WorkflowNode<any>['id'], WorkflowNode<any>>(
+		wf.value.nodes.map((node) => [node.id, node])
+	);
+	const nodeCache = new Map<WorkflowOutput<any>['id'], WorkflowNode<any>[]>();
+	wf.value.edges.forEach((edge) => {
+		if (!edge.source || !edge.target) return;
+		if (!nodeCache.has(edge.source)) nodeCache.set(edge.source, []);
+		nodeCache.get(edge.source)?.push(nodeMap.get(edge.target) as WorkflowNode<any>);
+	});
```

Comment:
Could you add a comment on what is happening in this loop please?

Reply:
It is building a traversal map from the edges, per comment on line 722 - I am not sure how else to describe it. 
I could document a simple example if that helps.